server-spi/src/main/java/org/keycloak/models/UserModel.java

```diff
@@ -56,7 +59,13 @@
     // No default method here to allow Abstract subclasses where the username is provided in a different manner
     String getUsername();
 
-    // No default method here to allow Abstract subclasses where the username is provided in a different manner
+    /**
+     * Sets username for this user, username should be switched to lowercase before setting.
```

Comment:
It should not, there is demand for case-sensitive username support. See e.g. https://issues.redhat.com/browse/KEYCLOAK-5699.
For this moment, Keycloak uses case-insensitive usernames. Is this because of search? Then we need to stress that it needs to be case-insensitive searchable, but that will be done later in the search operation of the underlying storage.